### PR TITLE
Add Golang and "go test" presubmit to oss-test-infra

### PR DESCRIPTION
**Pull request by clarketm**
1. Init Golang module.
2. Add "go test" presubmit to `oss-test-infra`
3. Add go test to verify trusted jobs. 

/assign @cjwagner @fejta

### CODE REVIEW

File: prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml [hunk 1]
```diff
@@ -1,16 +1,30 @@
...
+        containers:
+          - image: gcr.io/k8s-prow/checkconfig:latest
+            imagePullPolicy: Always
+            command:
+              - /checkconfig
+            args:
+              - --config-path=./prow/config.yaml
+              - --job-config-path=./prow/prowjobs
+
+    - name: pull-prow-jobs-validate
```
[the remark this answers is not in the document]
Comment (clarketm) [reply prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml]: Good tip! Updated.

File: prow/tests/jobs_test.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 Google LLC
```
Comment (krzyzacy) [inline prow/tests/jobs_test.go]: nit: 2019

Comment (clarketm) [reply prow/tests/jobs_test.go]: 🤦‍♂ . Good catch! Updated.

**Comment by clarketm** [comment]
@krzyzacy - I am not familiar enough with `checkconfig` to say if this *trusted* cluster verification is better suited to live there. It sounds like a good proposal. @cjwagner / @stevekuznetsov  - any thoughts on adding trusted cluster validation logic to `checkconfig`.

File: prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml [hunk 3]
```diff
@@ -1,16 +1,30 @@
...
-        - --job-config-path=./prow/prowjobs
+
+    - name: pull-prow-config-validate
+      branches:
+        - master
+      decorate: true
+      always_run: true
+      spec:
+        containers:
+          - image: gcr.io/k8s-prow/checkconfig:latest
```
Comment (stevekuznetsov) [inline prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml]: You likely want to make this version match the deployed version of Prow that this config is used in, to make sure that you get valid checks.

Comment (clarketm) [reply prow/prowjobs/GoogleCloudPlatform/oss-test-infra/gcp-oss-test-infra-config.yaml]: 👍 . I was going to add it in a separate PR, but it is small enough to include here.

**Pull request merged**